packages/react/docs/guides/tech-stack.md

```diff
@@ -0,0 +1,283 @@
+```
+
+destructuring is a great addtion to the language that cleans up the code base.
+
+## [Spread snytax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) (...)
+
+allows an iterable such as an array expression or string to be expanded in places where zero or more arguments (for function calls) or elements (for array literals) are expected, or an object expression to be expanded in places where zero or more key-value pairs (for object literals) are expected.
+
+```js
+function sum(x, y, z) {
+  return x + y + z;
+}
+
+const numbers = [1, 2, 3];
+
+sum(...numbers);
+// expected return: 6
+```
+
+In the PAL project the most common use is adding items to an array. Instead of mutating an original array we can create a new one and simply spread the old.
+
+```js
+const arr = [1, 2, 3];
+
+const newArr = [...arr, 4, 5, 6];
+// expected return: 6
+```
+
+Another common use is to remove items from an object before passing it on to another component.
+
+```js
+const obj = { one: 1, two: 2, three: 3 };
+
+const { one, ...rest } = obj;
+...
+<div value={rest}>
+
+// rest being { two: 2, three: 3 }
+```
+
+## [Arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) () => {}
+
+are used through out the project when their compact style makes it much easier to read the code.
+
+```js
+// if only one argument you can even drop the parens
+setState((prevState) => !prevState);
+// vs
+setState(function (prevState) {
+  return !prevState;
+});
+```
+
+Information to keep in mind when dealing with Arrow functions:
+
+- basic syntax `() => {}`. The parens are always needed except...
+- When only one argument is used you can drop the parens. ex. `argument => { return argument++}`
+- Implicit return: Arrow functions can have either a "concise body" or the usual "block body". In a concise body, only an expression is specified, which becomes the implicit return value. ex. `argument => argument++`
+- Can not use `this` keyword like you would with traditional functions.
+- They do not have an `arguments` object. Both `arguments` and `this` reference the value of the enclosing lexical scope.
+
+There are crucial differences between arrow functions and traditional functional expressions so be aware and know [when to not use an arrow function](https://www.javascripttutorial.net/es6/when-you-should-not-use-arrow-functions/). With the above information in mind, I would lean towards using function delcaration for most instances and arrow functions when used as an argument in another function.
+
+When writing a functional component it is best practice to use a regular function declaration
+
+```js
+function MyComponent(props) {
+  return <div>Hello World</div>;
+}
+```
+
+but you will definitely see them defined using arrow functions both in our library and in the wild.
+
+```js
+const MyComponent = (props) => <div>Hello World</div>;
```

Comment:
A point to make here, is that while regular function is preferred, cases where you need wrap the function with some other HOC, the arrow function is preferred.
```js
const MyComponent = observer((props) => <div>Hello World</div>);
```